crates/sui-framework-build/src/compiled_package.rs

```diff
@@ -0,0 +1,273 @@
+    /// verifies this.
+    // TODO: replace this with actual versioning checks instead of hacky byte-for-byte comparisons
+    pub fn verify_framework_version(
+        &self,
+        ext_sui_framework: Vec<CompiledModule>,
+        ext_move_stdlib: Vec<CompiledModule>,
+    ) -> SuiResult<()> {
+        // We stash compiled modules in the Modules map which is sorted so that we can compare sets of
+        // compiled modules directly.
+        let ext_framework_modules = Modules::new(ext_sui_framework.iter());
+        let pkg_framework_modules = self.get_framework_modules();
+
+        // compare framework modules pulled as dependencies (if any - a developer may choose to use only
+        // stdlib) with framework modules bundled with the distribution
+        if !pkg_framework_modules.is_empty()
+            && ext_framework_modules != Modules::new(pkg_framework_modules)
+        {
+            // note: this advice is overfitted to the most common failure modes we see:
+            // user is trying to publish to testnet, but has a `sui` binary and Sui Framework
+            // sources that are not in sync. the first part of the advice ensures that the
+            // user's project is always pointing at the devnet copy of the `Sui` Framework.
+            // the second ensures that the `sui` binary matches the devnet framework
+            return Err(SuiError::ModuleVerificationFailure {
+            error: "Sui framework version mismatch detected.\
+		    Make sure that you are using a GitHub dep in your Move.toml:\
+		    \
+                    [dependencies]
+                    Sui = { git = \"https://github.com/MystenLabs/sui.git\", subdir = \"crates/sui-framework\", rev = \"devnet\" }
+`                   \
+                    If that does not fix the issue, your `sui` binary is likely out of date--try \
+                    cargo install --locked --git https://github.com/MystenLabs/sui.git --branch devnet sui"
+                .to_string(),
+        });
+        }
+
+        let ext_stdlib_modules = Modules::new(ext_move_stdlib.iter());
+        let pkg_stdlib_modules = self.get_stdlib_modules();
+
+        // compare stdlib modules pulled as dependencies (if any) with stdlib modules bundled with the
+        // distribution
+        if !pkg_stdlib_modules.is_empty() && ext_stdlib_modules != Modules::new(pkg_stdlib_modules)
+        {
+            return Err(SuiError::ModuleVerificationFailure {
+                error: "Move stdlib version mismatch detected.\
+                    Make sure that the sui command line tool and the Move standard library code\
+                    used as a dependency correspond to the same git commit"
+                    .to_string(),
+            });
+        }
+
+        Ok(())
+    }
+}
+
+impl Default for BuildConfig {
+    fn default() -> Self {
+        BuildConfig {
+            config: MoveBuildConfig::default(),
+            run_bytecode_verifier: false,
```

Comment:
might want to set this to true by default. It's good for catching bugs in the compiler (and hopefully isn't too slow)

Reply:
Good call! Especially because it also runs the Sui verifier, which the compiler does not try to avoid.